### PR TITLE
Bug 1833115 - Migrate to compose foundation insets

### PR DESCRIPTION
**Pull request by rahulsainani**
– Migrates to compose foundation insets, removing the deprecated accompanist insets library
– Using material `TopAppBar`, removing the need for insets ui library.
– Removes `CompositionLocalProvider` that was unused
– Extract title logic so the params passed to private `TopAppBar` are `val` for compose compiler to mark it stable most likely.

### Screenshots
| Before | After | 
|------|-------|
|before|after

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1833115

### CODE REVIEW

**Comment by iorgamgabriel** [comment]
If the app theme is dark the status bar doesn't look good.

File: focus-android/app/src/main/java/org/mozilla/focus/settings/BaseComposeFragment.kt [hunk 1]
```diff
@@ -70,51 +68,60 @@ abstract class BaseComposeFragment : Fragment() {
...
     }
+
+    private fun getTitle(): String {
+        var title = ""
+        titleRes?.let { title = getString(it) }
+        titleText?.let { title = it }
+        return title
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (rahulsainani) [reply focus-android/app/src/main/java/org/mozilla/focus/settings/BaseComposeFragment.kt:102]: We're just using the material `TopAppBar`, the function basically calls that, this PR just extracted that into a function

**Comment by rahulsainani** [comment]
@iorgamgabriel This should be fixed now ☝️ thanks @mcarare for the help there!

**Comment by iorgamgabriel** [comment]
It looks good ! Thank you

**Review by iorgamgabriel — APPROVED** [review]

**Review by mcarare — APPROVED** [review]

**Pull request merged**